api/v1alpha1/module_types.go

```diff
@@ -50,13 +50,14 @@ type PushOptions struct {
 }
 
 type Build struct {
-
 	// +optional
 	// BuildArgs is an array of build variables that are provided to the image building backend.
 	BuildArgs []BuildArg `json:"buildArgs"`
 
 	Dockerfile string `json:"dockerfile"`
 
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
why do we need NodeSelector in Build? Why should user care where on which node the build is running?

Reply:
because if you have a hybrid cluster with ARM and x86 nodes you need to compile your driver on the same arch as you're deploying onto.

Reply:
I think that in case of multi-arch architecture, KMMO should be the one that decides where to run the Build job and will know how to place it correctly.

Reply:
How do we determine the target architecture of a given image?

Reply:
Now using the `Module`'s top-level `selector`.

Reply:
So are we keeping the `NodeSelector`? The other (and more automated way) will be to:
* use podman to build and release multi-arch images, then, it doesn't matter where the build is scheduled, the correct manifest of the image will be chosen.

This will require additional update to DTK's documentation, and need to be validated.